interfaces/builtin/network_manager.go

```diff
@@ -88,8 +89,10 @@ network packet,
 # Allow read and write access for all netplan configuration files
 # as NetworkManager will start using them to store the network
 # configuration instead of using its own internal keyfile based
-# format.
+# format. We also allow calling netplan so we can make the changes
+# effective without needing to re-start the device.
 /etc/netplan/{,**} rw,
+/usr/sbin/netplan Ux,
```

Comment:
I don't particularly care for this one since a bug in netplan or a future feature/something we don't know about might allow the snap to escape confinement. No other interfaces have Ux rules (except classic-support, which is highly specialized). I suggest instead picking up https://github.com/snapcore/snapd/pull/5915 and have the network-manager snap plugs network-setup-control.

Reply:
Following the discussion in https://github.com/snapcore/snapd/pull/5915